crowbar_framework/app/models/node_object.rb

```diff
@@ -1451,6 +1451,26 @@ def disk_release!(device, owner)
     disk_release(device, owner) and save
   end
 
+  def check_unclaimed_disks
+    unclaimed_disks = Array.new
+    crowbar_wall[:claimed_disks].each do |disk, _claim|
+      path, device_name = disk.split("/")[-2..-1]
+
+      unless disk =~ /^\/dev\/disk\//
+        unclaimed_disks.push(device_name) unless node[:block_device][path].nil?
```

Comment:
I guess this is supposed to handle disks claimed by name directly in /dev (i.e. /dev/sda, /dev/vdb and so on)?

I am pretty sure something is wrong with that ```unless``` condition. ```path``` will contain ```dev``` in such a case but you'd need to check for ```device_name``` here (which would be ```sda``` or something like that).

And then you only want to add such a disk to ```unclaimed_disks``` if it does not exist in ```node[:block_device]```.
So I think this should be:

```unclaimed_disks.push(device_name) if node[:block_device][device_name].nil?```